### PR TITLE
AMP hello world

### PR DESCRIPTION
**Pull request by nicl**
## What does this change?

Adds a minimal 'hello world' AMP page, including a simple test.

## Why?

This is our initial milestone with AMP!

Note: have borrowed a lot from https://github.com/guardian/dotcom-rendering/pull/228 - thanks @AWare for the great work on this! 

The changes are to:

* keep things scoped to just hello world (this is the main change and reason I thought it would be easier to open a new PR)
* to keep all AMP code under an /amp directory so that it stays truly separate
* add a basic test using the Google AMP validation library
* change some naming conventions

@SiAdcock and others, I am sure you have thoughts on how to implement Jest/testing properly. It would be great to not block this on that (unless absolutely necessary - in which case I'll just drop the testing commit), while realising that we may have to redo our tests to reflect whatever approach is chosen by the dev ex stream! But equally, if you have strong views already, I'm happy to update to reflect those.

### CODE REVIEW

File: packages/rendering/server.ts [hunk 1]
```diff
@@ -36,8 +37,23 @@ const renderArticle = ({ body }: express.Request, res: express.Response) => {
...
+        res.status(200).send(resp);
+    } catch (e) {
+        res.status(500).send(`<pre>${e.stack}</pre>`);
+    }
+};
+
 // this export is the function used by webpackHotServerMiddleware in /dev-server.js
-export default () => renderArticle;
+export default (options: any) => {
+    if ('amp' in options) return renderAMPArticle;
```
Comment (nicl) [inline packages/rendering/server.ts]: This should be the only `if (amp)` allowed! 😬

File: frontend/package.json [hunk 2]
```diff
@@ -17,23 +17,47 @@
     "emotion": "^9.1.1",
     "emotion-server": "^9.1.1",
     "html-minifier": "^3.5.14",
+    "jest": "^23.6.0",
```
Comment (SiAdcock) [inline frontend/package.json]: Perhaps jest and ts-jest should be devDependencies?

**Comment by SiAdcock** [comment]
Jest is a safe bet for test library of choice 👍

File: frontend/package.json (hunk 2 above)
Comment (nicl) [reply frontend/package.json]: Oh wait in addition I somehow added jest twice. Will tidy this up thanks. 👍

**Pull request merged**